### DISCUSSION TITLE
New Language/BCL Features with LTS releases

### DISCUSSION THREAD

**Discussion by iancooper** [BrighterCommand/Brighter#1570] [General]
I just wanted to note, for feedback, that my working model is that we use new language features, BCL classes etc. with an LTS release of .NET.

So we should currently target .NET Core 3.1 for our feature set, as it is LTS. We will next target .NET 6, which will also be LTS.

If C# depends on a version of .NET, then we won't use those language features, until the .NET LTS hits a version that supports them. So we are not using C# 9 features, because they are not supported by .NET Core 3.1. We can support those language features when we move to .NET 6.

In the end we need a strategy, and this seems to be the most predictable for others as we track a well-advertised MS policy.

**Comment by preardon** [comment]
Completely agree with this approach. I know I enjoying using the new and shiny things, however for packages such as Brighter, aligning to features of LTS releases and giving users more time to get their stuff aligned appears sensible.
